thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/algorithm/LegacyAlertFilterWrapper.java

```diff
@@ -0,0 +1,98 @@
+package com.linkedin.thirdeye.detection.algorithm;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.linkedin.thirdeye.datalayer.dto.DetectionConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.detection.ConfigUtils;
+import com.linkedin.thirdeye.detection.DataProvider;
+import com.linkedin.thirdeye.detection.DetectionPipeline;
+import com.linkedin.thirdeye.detection.DetectionPipelineResult;
+import com.linkedin.thirdeye.detector.email.filter.BaseAlertFilter;
+import com.linkedin.thirdeye.detector.email.filter.DummyAlertFilter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.apache.commons.collections.MapUtils;
+
+
+/**
+ * The Legacy alert filter wrapper. This wrapper runs the legacy alert filter in detection pipeline and filter out
+ * the anomalies that don't pass the filter.
+ */
+public class LegacyAlertFilterWrapper extends DetectionPipeline {
+  private static final String PROP_LEGACY_ALERT_FILTER_CLASS_NAME = "legacyAlertFilterClassName";
+  private static final String PROP_NESTED = "nested";
+  private static final String PROP_CLASS_NAME = "className";
+  private static final String PROP_ALERT_FILTER_LOOKBACK = "alertFilterLookBack";
+  private static String PROP_SPEC = "specs";
+  private static String PROP_ALERT_FILTER = "alertFilter";
+
+
+  private BaseAlertFilter alertFilter;
+  private final List<Map<String, Object>> nestedProperties;
```

Comment:
Random thought: could we inject a "LegacyMergeWrapper" property here by default? (and then the LegacyDimensionWrapper inside the LegacyMergeWrapper, and so on)
This way, user can still specify any kind of workflow but can safe the effort if they only want a single filter-merge-dimension-algo flow.
Obviously not related to the PR specifically.